client/tso_client.go

```diff
@@ -64,6 +63,13 @@ var tsoReqPool = sync.Pool{
 	},
 }
 
+func (req *tsoRequest) tryDone(err error) {
```

Comment:
why need to try? from the goroutine we can make sure here is no blocking.

Reply:
A request may be revoked right after it is sent to `tsoBatchController`, then if we double-check the contexts and return an error, `done <- err` will be blocked.